src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -0,0 +1,192 @@
+  * If one is using file editor gedit, then tab width has to be 2 spaces with use spaces option enabled. 
+  * Please don't edit anything beyond the space alloted to you otherwise we might get into a messy merge conflict.
+  * 
+  * For your two methods and their javadocs, 90 lines of code space(line no 38 - line no 127) is alloted to you. 
+  * In case A) you need more lines to be alloted or B) there is any need to make some changes outside the alloted space, 
+  * A pull request has to be made on my branch.
+  * I'll merge your branch and A) modify my branch to make requested changes or B) give some more lines of code space in my branch. 
+  * You need to then fetch my branch and again make a pull request on my branch for final merge.
+  */
+  //Manas's code starts from here.
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  // Manas's code finishes here
+
+
+
+  void runAll(){
+  
+    //get all the test cases in infrastucture module
+ 	  File testFolderInfrastucture = new File("../../../infrastucture/tests");
+	  File[] listOfTests = testFolder.listFiles();
+	  System.out.println("Running all the test cases of infrastucture module...");
+	  for (File file : listOfTests){
+    if (file.isFile()) {
+        runByName(file.getName());
+    }
+    
+    //get all the test cases in networking module
+ 	  File testFolderInfrastucture = new File("../../../networking/tests");
+	  File[] listOfTests = testFolder.listFiles();
+	  System.out.println("Running all the test cases of networking module...");
+	  for (File file : listOfTests){
+    if (file.isFile()) {
+        runByName(file.getName());
+    }
+    
+    //get all the test cases in processing module
+ 	  File testFolderInfrastucture = new File("../../../processing/tests");
+	  File[] listOfTests = testFolder.listFiles();
+	  System.out.println("Running all the test cases of processing module...");
+	  for (File file : listOfTests){
+    if (file.isFile()) {
+        runByName(file.getName());
+    }
+    
+    //get all the test cases in UI module
+ 	  File testFolderInfrastucture = new File("../../../UI/tests");
+    File[] listOfTests = testFolder.listFiles();
+	  System.out.println("Running all the test cases of UI module...");
+	  for (File file : listOfTests){
+    if (file.isFile()) {
+        runByName(file.getName());
+    }
+    
+}
+	  
+  }  
+  
+ /**
+  * Boolean method to give result of the Test Case Class
+  * @param testName    The file name of test case class including .java extension
+  * @return    <code>true</code> if the test case pass 
+  *     and actual output is same as expected output; 
+  *     <code>false</code> otherwise.
+  */
+  public boolean runByName(String testName){
+    String[] arrOfStr = str.split(".", 2); 
```

Comment:
What is 'str'? If you are trying to split test name, you need to split it as [Test name] and "Test". There won't be any punctuations. We have removed the underscore naming too.